### PR TITLE
[TACHYON-1326] Add allowExists option to mkdir

### PR DESCRIPTION
**Pull request by aaudiber**
https://tachyon.atlassian.net/browse/TACHYON-1326

### CODE REVIEW

File: servers/src/main/java/tachyon/master/file/meta/InodeTree.java [hunk 1]
```diff
@@ -283,8 +283,11 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
         traversalResult.getNonPersisted().add(lastInode);
         toPersistDirectories.add(lastInode);
       } else {
-        LOG.info(ExceptionMessage.FILE_ALREADY_EXISTS.getMessage(path));
-        throw new FileAlreadyExistsException(ExceptionMessage.FILE_ALREADY_EXISTS.getMessage(path));
+        if (!(lastInode.isDirectory() && options.isAllowExists())) {
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/file/meta/InodeTree.java]: NIT: else if?

Comment (yupeng9) [inline servers/src/main/java/tachyon/master/file/meta/InodeTree.java]: Shouldn't the condition be `lastInode.isDirectory() && !options.isAllowExists()`?

Comment (aaudiber) [inline servers/src/main/java/tachyon/master/file/meta/InodeTree.java]: Changed to `else if`, thanks

**Comment by yupeng9** [comment]
LGTM. Thanks!

File: clients/unshaded/src/test/java/tachyon/client/file/options/MkdirOptionsTest.java [hunk 2]
```diff
@@ -30,14 +30,12 @@
   @Test
   public void builderTest() {
     Random random = new Random();
+    boolean allowExists = random.nextBoolean();
     boolean recursive = random.nextBoolean();
     UnderStorageType ufsType = UnderStorageType.SYNC_PERSIST;
 
-    MkdirOptions options =
```
Comment (jsimsa) [inline clients/unshaded/src/test/java/tachyon/client/file/options/MkdirOptionsTest.java]: nit: the individual setters were on separate lines on purpose here to improve readability

File: servers/src/test/java/tachyon/master/file/meta/options/CreatePathOptionsTest.java [hunk 3]
```diff
@@ -35,16 +36,12 @@ public void builderTest() {
     boolean recursive = random.nextBoolean();
     long ttl = random.nextLong();
 
-    CreatePathOptions options =
```
Comment (jsimsa) [inline servers/src/test/java/tachyon/master/file/meta/options/CreatePathOptionsTest.java]: ditto

File: servers/src/test/java/tachyon/master/file/options/MkdirOptionsTest.java [hunk 4]
```diff
@@ -26,17 +26,16 @@
   @Test
   public void builderTest() {
     Random random = new Random();
+    boolean allowExists = random.nextBoolean();
     long operationTimeMs = random.nextLong();
     boolean persisted = random.nextBoolean();
     boolean recursive = random.nextBoolean();
 
-    MkdirOptions options =
```
Comment (jsimsa) [inline servers/src/test/java/tachyon/master/file/options/MkdirOptionsTest.java]: ditto

**Comment by jsimsa** [comment]
LGTM

File: clients/unshaded/src/test/java/tachyon/client/file/options/MkdirOptionsTest.java (hunk 2 above)
Comment (aaudiber) [inline clients/unshaded/src/test/java/tachyon/client/file/options/MkdirOptionsTest.java]: fixed, thanks

File: servers/src/test/java/tachyon/master/file/meta/options/CreatePathOptionsTest.java (hunk 3 above)
Comment (aaudiber) [inline servers/src/test/java/tachyon/master/file/meta/options/CreatePathOptionsTest.java]: fixed, thanks

File: servers/src/test/java/tachyon/master/file/options/MkdirOptionsTest.java (hunk 4 above)
Comment (aaudiber) [inline servers/src/test/java/tachyon/master/file/options/MkdirOptionsTest.java]: fixed, thanks

**Comment by jsimsa** [comment]
still LGTM

**Pull request merged**